src/products/components/ProductVariantPage/ProductVariantPage.tsx

```diff
@@ -149,6 +152,14 @@ const ProductVariantPage: React.FC<ProductVariantPageProps> = ({
   const [isModalOpened, setModalStatus] = React.useState(false);
   const toggleModal = () => setModalStatus(!isModalOpened);
 
+  const [
+    isEndPreorderModalOpened,
+    setIsEndPreorderModalOpened
+  ] = React.useState(false);
+
+  const togglePreorderModal = () =>
```

Comment:
The actions below indicate an exact outcome like "onClose" or "onEndPreorderTrigger" - if they never intend to toggle the modal on/off, I'd suggest to be explicit:
```
onClose={() => setIsEndPreorderModalOpened(false)}
// etc
```

Reply:
Good point - changed